source/slang/slang-check-impl.h

```diff
@@ -2652,7 +2652,10 @@ struct SemanticsVisitor : public SemanticsContext
 
     /// Compare items `left` and `right` produced by lookup, to see if one should be favored for
     /// overloading.
-    int CompareLookupResultItems(LookupResultItem const& left, LookupResultItem const& right);
+    int CompareLookupResultItems(
+        LookupResultItem const& left,
+        LookupResultItem const& right,
+        bool deferExtensionPrefs = false);
```

Comment:
instead of adding a `bool` parameter here, we should introduce a `LookupResultComparisonFlavor` enum that is either `BeforeOverloadResolution` or `AfterOverloadResolution`.

Then we enable the extension related rules only when `flavor == LookupResultComparisonFlavor::AfterOverloadResolution`.